papers/opencadd-klifs/paper.md

```diff
@@ -0,0 +1,62 @@
+bibliography: paper.bib
+---
+
+# Summary
+
+Protein kinases are involved in most aspects of cell life due to their role in signal transduction. Dysregulated kinases can cause severe diseases such as cancer, inflammatory and neurodegenerative diseases, which has made them a frequent target in drug discovery for the last decades [@Cohen:2021].
+The immense research on kinases has led to an increasing amount of kinase resources [@Kooistra:2017].
+Among them is the KLIFS database, which focuses on storing and analyzing structural data on kinases and interacting drugs and other small molecules [@Kanev:2021].
+The OpenCADD-KLIFS Python module offers a convenient integration of the KLIFS data into workflows to facilitate computational kinase research.
+
+# Statement of need
+
+[OpenCADD-KLIFS](https://opencadd.readthedocs.io/en/latest/databases_klifs.html) (``opencadd.databases.klifs``) is a part of the [OpenCADD](https://opencadd.readthedocs.io/) package, a collection of Python modules for structural cheminformatics.
+This module offers access to KLIFS data [@Kanev:2021] such as information about kinases, structures, ligands, 
+interaction fingerprints, and bioactivities. 
+KLIFS thereby focuses especially on the ATP binding site, defined as a set of 85 residues and aligned across all structures using a multiple sequence alignment (MSA) [@vanLinden:2014].
+With OpenCADD-KLIFS, KLIFS data can be queried either locally from a KLIFS download or remotely from the KLIFS webserver. 
+The presented module provides identical APIs for the remote and local queries for KLIFS data and streamlines all output into 
+standardized Pandas DataFrames [@pandas] to allow for easy and quick downstream data analyses (\autoref{fig:opencadd_klifs_toc}). This Pandas-focused setup is ideal to work with in Jupyter notebooks [@Kluyver:2016]. 
+
+
+![OpenCADD-KLIFS fetches KLIFS data [@Kanev:2021] offline from a KLIFS download or online from the KLIFS database and formats the output as user-friendly Pandas DataFrames [@pandas].\label{fig:opencadd_klifs_toc}](opencadd_klifs_toc.png)
+
+The KLIFS database offers a REST API including a Swagger API specification [@klifsswagger]. Our module OpenCADD-KLIFS uses bravado [@bravado] to dynamically generate a Python client based on these Swagger API definitions and adds wrappers to enable the following functionalities:
+
+- A session is set up, which allows access to various KLIFS *data sources* by different *identifiers* with the API ``session.data_source.by_identifier``. *Data sources* currently include kinases, structures and annotated conformations, modified residues, pockets, ligands, drugs, and bioactivities; *identifiers* refer to kinase names, PDB IDs, KLIFS IDs, and more.
+For example, ``session.structures.by_kinase_name`` fetches information on all structures for a query kinase.
+- Query results obtained from the remote KLIFS webserver are streamlined with those obtained from a local KLIFS download using the same API.
```

Comment:
Do you mean that the returned data follows the same _schema_ regardless the source (local/remote)?

Reply:
I see I have mixed two thoughts here (same API and same results schema): 
> - Query results obtained from the remote KLIFS webserver are streamlined with those obtained from a local KLIFS download using the same API.

I have updated the text as follows:

> - A session is set up, which allows access to various KLIFS *data sources* by different *identifiers* with the API ``session.data_source.by_identifier``. *Data sources* currently include kinases, structures and annotated conformations, modified residues, pockets, ligands, drugs, and bioactivities; *identifiers* refer to kinase names, PDB IDs, KLIFS IDs, and more. For example, ``session.structures.by_kinase_name`` fetches information on all structures for a query kinase.
> - The same API is used for local and remote sessions.
> - The returned data follows the same schema regardless of the session type (local/remote); all results obtained with bravado are formatted as Pandas DataFrames with standardized column names, data types, and handling of missing data.

Reply:
Excellent!